mantidimaging/gui/windows/spectrum_viewer/spectrum_widget.py

```diff
@@ -42,8 +42,23 @@ def __init__(self, parent: 'SpectrumViewerWindowView'):
         self.range_control = LinearRegionItem()
         self.range_control.sigRegionChanged.connect(self._handle_tof_range_changed)
 
-        self.roi = ROI(pos=(0, 0), rotatable=False, scaleSnap=True, translateSnap=True)
-        self.roi.sigRegionChanged.connect(self.roi_changed.emit)
+        self.roi_dict: dict[str, SensibleROI] = {}  # hint: "roi_dict: Dict[<str>, <SensibleROI>]
+
+        self.max_roi_size = [0, 0]
+
+    def set_max_roi_size(self, width: int, height: int) -> None:
+        """
+        If width or height larger than max_roi_size,
+        set max_roi_size to higher width or height
+
+        :param width: width of the image
+        :param height: height of the image
+        """
+
+        if width > self.max_roi_size[0]:
+            self.max_roi_size[0] = width
+        if height > self.max_roi_size[1]:
+            self.max_roi_size[1] = height
```

Comment:
I agree with you about it not being the cleanest solution. I'll look into coming up with another solution. This was put in place as the "all" roi does not retain size for some reason which causes the roi to contain nan values only. I implemented the above to make sure that the roi with the largest size was used for setting the "all" roi size without creating an instance of roi in the spectrum_widget which would make the "all" roi visible in the GUI.

Reply:
I've reworked this slightly realising that `set_max_roi_size()` is not needed as the size of an ROI when initially adding to `self.roi_dict` will always be the same a the image size.
I do still set a max_roi size in the init as I can't see any other means of setting the size of the "all" roi without making it visible in the gui or passing in a SensibleROI through `get_roi()`to replicate the functionality of doing this using `self.roi` which was removed as part of this issue in favour of `self.roi_dict`. What are your thoughts on this?

Reply:
ok, that looks better now. thanks

Reply:
Actually, one more thing. Could you make it a specific case for when `roi_name` is `all`. So that if `get_roi` is called with a non existent name then there is an error. This will make it quicker to catch any bugs that show up one ROIs can be removed.